packages/tools/src/stateManagement/annotation/config/helpers.ts

```diff
@@ -0,0 +1,55 @@
+import { StyleSpecifications } from '../../../types/AnnotationStyle';
+import { ToolModes, AnnotationStyleStates } from '../../../enums';
+import toolStyle from './ToolStyle';
+
+/**
+ * Build a list of alternative property names in ascending order of priority
+ * @param property - The base property name -- e.g., 'color'
+ * @param state - An optional state to determine the final property name
+ * @param mode - An optional mode to determine the final property name
+ * @returns A list of alternative property names
+ */
+function getStyleAlternatives(
+  property: string,
+  state?: AnnotationStyleStates,
+  mode?: ToolModes
+): string[] {
+  const list = [`${property}`];
+  if (state) list.push(`${list[0]}${state}`);
+  if (mode) list.push(`${list[list.length - 1]}${mode}`);
+  return list;
+}
+
+/**
+ * Get the value of a style property from the ToolStyle config
+ * @param property - The name of the property to get.
+ * @param styleSpecifications - An object containing the specifications such as viewportId,
+ * toolGroupId, toolName and annotationUID which are used to get the style if the level of specificity is
+ * met (hierarchy is checked from most specific to least specific which is
+ * annotationLevel -> viewportLevel -> toolGroupLevel -> default.
+ * @param state - The state of the tool (Default, Locked etc.)
+ * @param mode - The current tool mode. (Active, Passive etc.)
+ * @returns The value of the property.
+ */
+function getStyleProperty(
+  property: string,
+  StyleSpecifications: StyleSpecifications,
+  state?: AnnotationStyleStates,
+  mode?: ToolModes
+): string {
+  // `alternatives` is a list of property names with priority in ascending
+  // order like: ['color', 'colorSelected', 'colorSelectedActive']
+  // Thus, we attempt resolving property names in reverse order
```

Comment:
We can avoid this reversing, and a comment why, if we just return them in the correct order from the helper, which I think would be so much simpler?

Reply:
it will include a lot of if/else statements based on the availability of the state and/or mode. I'll expand more on the comment and since it is internal I would say it is not that bad